packages/tx/src/eip7702Transaction.ts

```diff
@@ -0,0 +1,396 @@
+   * to sign the transaction (e.g. for sending to a hardware wallet).
+   *
+   * Note: in contrast to the legacy tx the raw message format is already
+   * serialized and doesn't need to be RLP encoded any more.
+   *
+   * ```javascript
+   * const serializedMessage = tx.getMessageToSign() // use this for the HW wallet input
+   * ```
+   */
+  getMessageToSign(): Uint8Array {
+    return EIP2718.serialize(this, this.raw().slice(0, 9))
+  }
+
+  /**
+   * Returns the hashed serialized unsigned tx, which can be used
+   * to sign the transaction (e.g. for sending to a hardware wallet).
+   *
+   * Note: in contrast to the legacy tx the raw message format is already
+   * serialized and doesn't need to be RLP encoded any more.
+   */
+  getHashedMessageToSign(): Uint8Array {
+    return EIP2718.getHashedMessageToSign(this)
+  }
+
+  /**
+   * Computes a sha3-256 hash of the serialized tx.
+   *
+   * This method can only be used for signed txs (it throws otherwise).
+   * Use {@link EOACodeEIP7702Transaction.getMessageToSign} to get a tx hash for the purpose of signing.
+   */
+  public hash(): Uint8Array {
+    return Legacy.hash(this)
+  }
+
+  /**
+   * Computes a sha3-256 hash which can be used to verify the signature
+   */
+  public getMessageToVerifySignature(): Uint8Array {
+    return this.getHashedMessageToSign()
+  }
+
+  /**
+   * Returns the public key of the sender
+   */
+  public getSenderPublicKey(): Uint8Array {
+    return Legacy.getSenderPublicKey(this)
+  }
+
+  addSignature(
+    v: bigint,
+    r: Uint8Array | bigint,
+    s: Uint8Array | bigint,
+    convertV: boolean = false
+  ): EOACodeEIP7702Transaction {
+    r = toBytes(r)
+    s = toBytes(s)
+    const opts = { ...this.txOptions, common: this.common }
+
+    return EOACodeEIP7702Transaction.fromTxData(
+      {
+        chainId: this.chainId,
+        nonce: this.nonce,
+        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+        maxFeePerGas: this.maxFeePerGas,
+        gasLimit: this.gasLimit,
+        to: this.to,
+        value: this.value,
+        data: this.data,
+        accessList: this.accessList,
+        authorizationList: this.authorizationList,
+        v: convertV ? v - BIGINT_27 : v, // This looks extremely hacky: @ethereumjs/util actually adds 27 to the value, the recovery bit is either 0 or 1.
+        r: bytesToBigInt(r),
+        s: bytesToBigInt(s),
+      },
+      opts
+    )
+  }
+
+  /**
+   * Returns an object with the JSON representation of the transaction
+   */
+  toJSON(): JsonTx {
+    const accessListJSON = AccessLists.getAccessListJSON(this.accessList)
+    const baseJson = super.toJSON()
+
+    return {
+      ...baseJson,
+      chainId: bigIntToHex(this.chainId),
+      maxPriorityFeePerGas: bigIntToHex(this.maxPriorityFeePerGas),
+      maxFeePerGas: bigIntToHex(this.maxFeePerGas),
+      accessList: accessListJSON,
+      authorizationList: this.AuthorizationListJSON,
```

Comment:
why are we saving json upfront in the

Reply:
We also do this in 1559 / 2929, I suggest we factor this out in another PR which also covers those txs